app/src/main/java/org/oppia/app/utility/Logger.kt

```diff
@@ -0,0 +1,197 @@
+   * @param msg The message you would like logged.
+   * @param tr  An exception to log
+   */
+  fun i(tag: String, msg: String, tr: Throwable) {
+    write(LogLevel.INFO, tag, msg, tr)
+  }
+
+  /**
+   *
+   * @param tag Used to identify the source of a log message.  It usually identifies
+   * the class or activity where the log call occurs.
+   * @param msg The message you would like logged.
+   */
+  fun w(tag: String, msg: String) {
+    write(LogLevel.WARNING, tag, msg)
+  }
+
+  /**
+   *
+   * @param tag Used to identify the source of a log message.  It usually identifies
+   * the class or activity where the log call occurs.
+   * @param msg The message you would like logged.
+   * @param tr  An exception to log
+   */
+  fun w(tag: String, msg: String, tr: Throwable) {
+    write(LogLevel.WARNING, tag, msg, tr)
+  }
+
+  /**
+   *
+   * @param tag Used to identify the source of a log message.  It usually identifies
+   * the class or activity where the log call occurs.
+   * @param msg The message you would like logged.
+   */
+  fun e(tag: String, msg: String) {
+    write(LogLevel.ERROR, tag, msg)
+  }
+
+  /**
+   *
+   * @param tag Used to identify the source of a log message.  It usually identifies
+   * the class or activity where the log call occurs.
+   * @param msg The message you would like logged.
+   * @param tr  An exception to log
+   */
+  fun e(tag: String, msg: String, tr: Throwable) {
+    write(LogLevel.ERROR, tag, msg, tr)
+  }
+
+  private fun isLogEnable(logLevel: LogLevel): Boolean {
+    return GLOBAL_LOG_LEVEL.logLevel < logLevel.logLevel
+  }
+
+  private fun write(logLevel: LogLevel, tag: String, log: String) {
+    if (isLogEnable(logLevel) && ENABLE_CONSOLE_LOG) {
+      Log.println(logLevel.logLevel, tag, log)
+    }
+    if (isLogEnable(logLevel) && ENABLE_FILE_LOG) {
+      val msg = Calendar.getInstance().time.toString() + "\t" + logLevel.name + "/" + tag + ": " + log
+      write(msg)
+    }
+  }
+
+  private fun write(logLevel: LogLevel, tag: String, log: String, tr: Throwable) {
+    if (isLogEnable(logLevel) && ENABLE_CONSOLE_LOG) {
+      Log.println(logLevel.logLevel, tag, log + "\n" + Log.getStackTraceString(tr))
+    }
+    if (isLogEnable(logLevel) && ENABLE_FILE_LOG) {
+      val msg =
+        Calendar.getInstance().time.toString() + "\t" + logLevel.name + "/" + tag + ": " + log + "\n" + Log.getStackTraceString(
+          tr
+        )
+      write(msg)
+    }
+  }
+
+  private fun write(text: String) {
+    var out: BufferedWriter? = null
+    var filePath = LOG_DIRECTORY
+    try {
+
+      val df = SimpleDateFormat("dd_MMM_yyyy", Locale.ENGLISH)
+      val formattedDate = df.format(System.currentTimeMillis())
+
+      if (!File(LOG_DIRECTORY).exists())
+        File(LOG_DIRECTORY).mkdir()
+
+      filePath = "$LOG_DIRECTORY$formattedDate.log"
+      if (!File(filePath).exists())
+        File(filePath).createNewFile()
+
+      val fStream = FileWriter(filePath, true)
```

Comment:
Could we instead have a long-lived print stream that we store at the class level? I think this will simplify a few things:
1. The directory doesn't need to be created since it's a system directory (if we use files dir)
2. The file doesn't need to be created since PrintStream should do that automatically
3. We don't need to open/close the stream for each write
4. These methods may be simplifiable in general once a print stream is being used

Reply:
Made use of PrintStream to write to a file